SpatialGDK/Source/SpatialGDKServices/Public/LocalDeploymentManager.h

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+#pragma once
+
+#include "Async/Future.h"
+#include "CoreMinimal.h"
+#include "FileCache.h"
+#include "Modules/ModuleManager.h"
+#include "Templates/SharedPointer.h"
+
+DECLARE_LOG_CATEGORY_EXTERN(LogSpatialDeploymentManager, Log, All);
+
+class FJsonObject;
+
+class FLocalDeploymentManager
+{
+public:
+	FLocalDeploymentManager();
+
+	void SPATIALGDKSERVICES_API RefreshServiceStatus();
+
+	bool SPATIALGDKSERVICES_API IsLocalDeploymentRunning();
+	bool SPATIALGDKSERVICES_API IsSpatialServiceRunning();
+
+	bool SPATIALGDKSERVICES_API TryStartLocalDeployment(FString LaunchConfig, FString LaunchArgs);
+	bool SPATIALGDKSERVICES_API TryStopLocalDeployment();
+
+	bool SPATIALGDKSERVICES_API TryStartSpatialService();
+	bool SPATIALGDKSERVICES_API TryStopSpatialService();
+
+	bool SPATIALGDKSERVICES_API GetLocalDeploymentStatus();
+	bool SPATIALGDKSERVICES_API GetServiceStatus();
+
+	bool SPATIALGDKSERVICES_API IsDeploymentStarting();
+	bool SPATIALGDKSERVICES_API IsDeploymentStopping();
+
+	bool SPATIALGDKSERVICES_API IsServiceStarting();
+	bool SPATIALGDKSERVICES_API IsServiceStopping();
+
+	// TODO: Refactor these into Utils
+	FString GetProjectName();
+	void WorkerBuildConfigAsync();
+	bool ParseJson(FString RawJsonString, TSharedPtr<FJsonObject>& JsonParsed);
+	void ExecuteAndReadOutput(FString Executable, FString Arguments, FString DirectoryToRun, FString& OutResult, int32& ExitCode);
```

Comment:
The non-out params should be const&, and you shouldn't be de-referencing them at all the calling sites. You're casting FString > char* > FString

Reply:
Very good points